### DISCUSSION TITLE
Teslamate install challenges on a Raspberry Pi 4 8GB - Help needed

### DISCUSSION THREAD

**Discussion by jebinc** [teslamate-org/teslamate#3082] [General]
Hello all!

![Screenshot 2023-01-21 at 3 52 27 PM](https://user-images.githubusercontent.com/123277557/213894712-f3914df8-510c-4797-b3ee-5ce35e2f031e.png)

I've successfully gotten to the point of creating the docker-compose.yml file, and per sets 2-3 I replaced the "#insert your secure database password!I" with a made up alphanumeric string.  Same for Encryption key.  Saved file, rebooted Pi and input my Pi's  "ip address:4000" and get the dreaded, "This site can't be reached" page.  When I enter Grafana, using "ip address:3000" I get a log on screen (email and password).  I thought maybe it was looking for my Tesla account email address and password, but no.

What am I doing wrong?

**Comment by cwanja** [comment]
Grafana is `admin` and `admin`. You can change it if it's exposed to the internet. 

For the key, I found it easier to simply have it be blank and then capture the key from the logs. Once you add the key to the YML, restart the containers. If you wish to use your own key, what does the `teslamate` container logs say? My guess is you might have an unapproved key or space in the line.

**Reply by jebinc** [reply]
Thanks for the quick reply!  Not sure how to access the container logs on a Raspberry Pi, as it's not my "go to" system, in general.  So, in the docker-compose.yml file I should delete all the rows (four) that contain "ENCRYPTION_KEY" and DATAPASS_PASS and POSTGRES_PASSWORD?  Again, I don't seem to have access to the Teslamate web screen (:4000), only the Grafana (:3000) one.  So, no way to link my Tesla account with Teslamate...

**Reply by cwanja** [reply]
Logs can be accessed by using a Terminal and running the following command in the same folder that your YML file is located.
`docker-compose logs teslamate`

I would say, try and access the logs from TeslaMate by the command above. If that does not work, shut down the application and remove just the contents of the encryption key line. See if that will load TeslaMate and allow you to log in.

**Reply by jebinc** [reply]
When I open the logs, I see about 20 rows that repeat the following two lines over and over...

"Teslamate_1 | waiting for Postgres at database:5432"
"Teslamate_1 | sleep: cannot read realtime clock:Operation not permitted"

**Reply by jebinc** [reply]
Any other suggestions?  Tried both above with no luck.

**Reply by cwanja** [reply]
What do the database logs say? Replace `teslamate` with the database service name from your YML file. 

Saw your other post. Have had no issues for almost four years on a Pi 3b+. Cannot imagine major changes between the 3 and 4 Pis that would render TeslaMate not functional.

**Reply by jebinc** [reply]
I decided to stop wasting time troubleshooting and reimaged the pi and started over.  Worked fine.  Something must have been hosed during the update process that was not easily resolved.
